build.gradle

```diff
@@ -107,23 +134,56 @@ List<String> resolveIntelliJVersions(String versionKey) {
   return [versionValue]
 }
 
-class ResolveIntelliJVersions extends DefaultTask {
-  @Input
-  @Option(option = 'version-key', description = 'Either release number (like 2020.3) or key of intellijVersions (like eapOfLatestSupportedMajor)')
-  String versionKey
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import org.jsoup.nodes.Document
+
+String findNewestEapBuildNumber() {
+  Document htmlContent = Jsoup.connect(project.intellijSnapshotsUrl).get()
+  Element link = htmlContent.select('a[href^='+project.intellijSnapshotsUrl+'com/jetbrains/intellij/idea/ideaIC/][href$="-EAP-SNAPSHOT.pom"]')[0]
```

Comment:
Oh! I did a quick experiment here... replaced `latestStable` in properties with `2021.3.3` (deliberately too low a version, so that some 2022.1 EAP gets picked up) and ended up with:

```
eapOfLatestSupportedMajor=221-EAP-SNAPSHOT
```

While this is technically a correct value... see `// Note that we have to use a "fixed snapshot"` comment in build.gradle!

I'd refine the regexes to only capture `X.Y.Z-EAP-SNAPSHOT`-style build numbers, and not `X-EAP-SNAPSHOT` ones